include/ddc/kernels/splines/bsplines_non_uniform.hpp

```diff
@@ -108,92 +115,184 @@ class NonUniformBSplines : detail::NonUniformBSplinesBase
                 ddc::DSpan2D derivs,
                 ddc::Coordinate<Tag> const& x,
                 std::size_t n) const;
 
+        /** @brief Compute the integrals of the B-splines.
+         *
+         * @param[out] int_vals The values of the integrals. It has to be a (1+nbasis) 1D mdspan.
+         * @return The values of the integrals.
+         */
         template <class Layout, class MemorySpace2>
         KOKKOS_INLINE_FUNCTION ddc::ChunkSpan<double, discrete_domain_type, Layout, MemorySpace2>
         integrals(
                 ddc::ChunkSpan<double, discrete_domain_type, Layout, MemorySpace2> int_vals) const;
 
+        /** @brief Returns the coordinate of the knot corresponding to the given index for a B-spline.
+         *
+         * @param[in] knot_idx Integer identifying index of the knot.
+         * @return Coordinate of the knot.
+         */
         KOKKOS_INLINE_FUNCTION ddc::Coordinate<Tag> get_knot(int knot_idx) const noexcept
         {
             return ddc::coordinate(ddc::DiscreteElement<mesh_type>(knot_idx + degree()));
         }
 
+        /** @brief Returns the coordinate of the first support knot associated to a discrete_element identifying a B-spline.
+         *
+         * @param[in] ix DiscreteElement identifying the B-spline.
+         * @return Coordinate of the knot.
+         */
         KOKKOS_INLINE_FUNCTION ddc::Coordinate<Tag> get_first_support_knot(
                 discrete_element_type const& ix) const
         {
             return ddc::coordinate(ddc::DiscreteElement<mesh_type>(ix.uid()));
         }
 
+        /** @brief Returns the coordinate of the last support knot associated to a discrete_element identifying a B-spline.
+         *
+         * @param[in] ix DiscreteElement identifying the B-spline.
+         * @return Coordinate of the knot.
+         */
         KOKKOS_INLINE_FUNCTION ddc::Coordinate<Tag> get_last_support_knot(
                 discrete_element_type const& ix) const
         {
             return ddc::coordinate(ddc::DiscreteElement<mesh_type>(ix.uid() + degree() + 1));
         }
 
+        /** @brief Returns the coordinate of the \f$n\f$-th knot associated to a DiscreteElement identifying a B-spline.
+         *
+         * @param[in] ix DiscreteElement identifying the B-spline.
+         * @param[in] n Integer indexing a knot in the support of the B-spline.
+         * @return Coordinate of the knot.
+         */
         KOKKOS_INLINE_FUNCTION ddc::Coordinate<Tag> get_support_knot_n(
                 discrete_element_type const& ix,
                 int n) const
         {
             return ddc::coordinate(ddc::DiscreteElement<mesh_type>(ix.uid() + n));
         }
 
+        /** @brief Returns the coordinate of the lower boundary knot.
+         *
+         * @return Coordinate of the lower boundary knot.
+         */
         KOKKOS_INLINE_FUNCTION ddc::Coordinate<Tag> rmin() const noexcept
```

Comment:
This is a bad description. We are knot assured that the knots are all inside the domain. I use the knot to calculate this but I don't think they should appear in the description.
```suggestion
        /** @brief Returns the coordinate of the lower bound of the domain on which the B-splines are defined.
         *
         * @return Coordinate of the lower bound of the domain.
         */
        KOKKOS_INLINE_FUNCTION ddc::Coordinate<Tag> rmin() const noexcept
```